lib/rules/no-misleading-character-class.js

```diff
@@ -232,11 +332,27 @@ module.exports = {
                     }];
                 }
 
-                context.report({
-                    node,
-                    messageId: kind,
-                    suggest
-                });
+                const locs = getNodeReportLocations(matches, node);
+
+                // Grapheme zero-width joiners (e.g. in 👨‍👩‍👦) visually show as one emoji
+                if (kind === "zwj" && locs.length > 1) {
+                    context.report({
+                        loc: {
+                            start: locs[0].start,
+                            end: locs[1].end
+                        },
+                        messageId: kind,
+                        suggest
+                    });
```

Comment:
This wouldn't show all if there are more of these in the pattern, but we can fix that in a follow-up.